### DISCUSSION TITLE
Intelligent tariff bug?

### DISCUSSION THREAD

**Discussion by Psai** [BottlecapDave/HomeAssistant-OctopusEnergy#711] [General]
Didn't know if this was a bug, or intended behaviour.

I'm on intelligent octopus, but since the last major update, the intelligent dispatching sensor hasn't turned on (not charged the car) and the off peak sensor is unavailable, so I have to rely on the cheap time window being hard set in my automations now instead of being triggered by sensors.

Am I missing something, or is this a bug?

**Comment by BottlecapDave** [comment]
Hello. This all _depends_...

* v10.0.0 saw intelligent sensors removed for people on intelligent tariff with a ohme charger (as these settings were wrongly exposed). This was in the changelog as a breaking change
* v10.0.2 saw intelligent sensors being removed for people on an Intelligent tariff, but not one where the sensors should have been exposed (e.g. Flux)
* v10.0.3 fixed intelligent sensors incorrectly being removed for people on certain Intelligent Go tariffs

The off peak sensor should always be available, but won't turn on for certain tariffs.

Do you have any errors in your logs which might indicate why you're experiencing this issue?

**Reply by Psai** [reply]
Thanks for the reply. 
I just checked and found this one which has occurred a lot.
I also tried removing and reconfiguring the octopus integration this morning, with no luck, so these 228 occurrences are since this morning).


<details><summary>custom_components.octopus_energy.coordinators.electricity_rates</summary>
<p>
This error originated from a custom integration.

Logger: custom_components.octopus_energy.coordinators.electricity_rates
Source: helpers/update_coordinator.py:300
Integration: Octopus Energy (documentation, issues)
First occurred: 07:30:01 (228 occurrences)
Last logged: 09:23:01

Unexpected error fetching ELECTRICITY_RATES_(removed)2 data: 'NoneType' object has no attribute 'completed'
Unexpected error fetching ELECTRICITY_RATES_(removed)2 data: 'NoneType' object has no attribute 'completed'
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 300, in _async_refresh
    self.data = await self._async_update_data()
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 256, in _async_update_data
    return await self.update_method()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/octopus_energy/coordinators/electricity_rates.py", line 133, in async_update_electricity_rates_data
    dispatches.dispatches.completed
AttributeError: 'NoneType' object has no attribute 'completed'

</p>
</details>

**Reply by BottlecapDave** [reply]
What tariff are you on?

**Reply by Psai** [reply]
![Screenshot_20240120-095024](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/3990521/cc7f97ad-4f12-4f53-b113-0f02c300f3aa)
Intelligent octopus go.


I'm just downloading 10.0.3, so I'll see if the issue is fixed (it literally just became available)

**Reply by Psai** [reply]
It looks like 10.0.3 has fixed it. Well done 👍
